chain/indexer/src/streamer/streamer.rs

```diff
@@ -92,12 +94,87 @@ async fn fetch_block_by_height(
         .map_err(|err| FailedToFetchData::String(err))
 }
 
+/// Fetches specific block by it's hash
+async fn fetch_block_by_hash(
+    client: &Addr<near_client::ViewClientActor>,
+    hash: CryptoHash,
+) -> Result<views::BlockView, FailedToFetchData> {
+    client
+        .send(near_client::GetBlock(near_primitives::types::BlockReference::BlockId(
+            near_primitives::types::BlockId::Hash(hash),
+        )))
+        .await?
+        .map_err(|err| FailedToFetchData::String(err))
+}
+
+async fn convert_transactions_sir_into_local_receipts(
+    client: &Addr<near_client::ViewClientActor>,
+    near_config: &neard::NearConfig,
+    txs: Vec<&IndexerTransactionWithOutcome>,
+    block: &views::BlockView,
+) -> Result<Vec<views::ReceiptView>, FailedToFetchData> {
+    let prev_block = fetch_block_by_hash(&client, block.header.prev_hash.clone()).await?;
+    let prev_block_gas_price = prev_block.header.gas_price;
+
+    let local_receipts: Vec<views::ReceiptView> = txs
+        .into_iter()
+        .map(|tx| {
+            let cost = tx_cost(
+                &near_config.genesis.config.runtime_config.transaction_costs,
+                &near_primitives::transaction::Transaction {
+                    signer_id: tx.transaction.signer_id.clone(),
+                    public_key: tx.transaction.public_key.clone(),
+                    nonce: tx.transaction.nonce,
+                    receiver_id: tx.transaction.receiver_id.clone(),
+                    block_hash: block.header.hash,
+                    actions: tx
+                        .transaction
+                        .actions
+                        .clone()
+                        .into_iter()
+                        .map(|action| {
+                            near_primitives::transaction::Action::try_from(action).unwrap()
+                        })
+                        .collect(),
+                },
+                prev_block_gas_price,
+                true,
+                near_config.genesis.config.protocol_version,
+            );
```

Comment:
It is not beautiful, but we should embrace it as is. At least, it is working correctly and consistently with the internal implementation.